contracts/statechannels/AllocationExchange.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/cryptography/ECDSA.sol";
+
+import "../governance/Governed.sol";
+import "../staking/IStaking.sol";
+import "../token/IGraphToken.sol";
+
+/**
+ * @title Allocation Exchange
+ * @dev This contract holds tokens that anyone with a voucher signed by the
+ * authority can redeem. The contract validates if the voucher presented is valid
+ * and then sends tokens to the Staking contract by calling the collect() function
+ * passing the voucher allocationID. The contract enforces that only one voucher for
+ * an allocationID can be redeemed.
+ * Only governance can change the authority.
+ */
+contract AllocationExchange is Governed {
+    // An allocation voucher represents a signed message that allows
+    // to redeem an amount of funds from this contract and collect
```

Comment:
Minor grammar: "allows to redeem" -> "allows redeeming"

Reply:
+1